tests/unit/test_footprinter.cc

```diff
@@ -474,48 +515,21 @@ TEST_F(TestFootprinter, test_vector) {
       sizeof(v)
     );
   }
-}
-
-struct TestBase {
-
-  checkpoint_virtual_serialize_root()
-
-  virtual ~TestBase() = default;
-
-  template <typename SerializerT>
-  void serialize(SerializerT& s) {
-    s | a;
-  }
-
-private:
-  int a = 0;
-};
-
-struct TestDerived2 : TestBase {
-  explicit TestDerived2(int i) {}
-  explicit TestDerived2(SERIALIZE_CONSTRUCT_TAG) {}
 
-  checkpoint_virtual_serialize_derived_from(TestBase)
+  {
+    std::vector<TestDerived2> v;
+    EXPECT_GE(
+      checkpoint::getMemoryFootprint(v),
+      sizeof(v)
+    );
 
-  template <
-    typename SerializerT,
-    typename = std::enable_if_t<
-      std::is_same<
-        SerializerT,
-        checkpoint::Footprinter
-      >::value
-    >
-  >
-  void serialize(SerializerT& s) {
-    s | raw_pointer;
-    s | shared_pointer;
+    v.reserve(3);
+    EXPECT_EQ(
+      checkpoint::getMemoryFootprint(v),
+      sizeof(v) + v.capacity() * sizeof(TestDerived2)
```

Comment:
ok, to avoid surprises, this could also fail on some platforms (if capacity was bigger than 3 by default):
> reserve() cannot be used to reduce the capacity of the container; to that end shrink_to_fit() is provided. 

`shrink_to_fit()` incoming

Reply:
It's not just if capacity is greater than 3 by default, it's also if the `reserve` call gives back a capacity greater than 3. This could happen if it rounds to some unit that it knows the allocator works in, for instance, or if the library implements http://wg21.link/p0401